userspace/libscap/engine/gvisor/scap_gvisor.cpp

```diff
@@ -377,4 +452,161 @@ int32_t engine::next(scap_evt **pevent, uint16_t *pcpuid)
     return SCAP_TIMEOUT;
 }
 
+std::vector<std::string> engine::runsc(char *argv[])
+{
+	std::vector<std::string> res;
+	int pipefds[2];
+
+	int ret = pipe(pipefds);
+	if(ret)
+	{
+		return res;
+	}
+
+	pid_t pid = vfork();
+	if(pid > 0)
+	{
+		char line[max_line_size];
+		int status;
+		
+		::close(pipefds[1]);
+		wait(&status);
+		if(status)
+		{
+			return res;
+		}
+
+		FILE *f = fdopen(pipefds[0], "r");
+		if(!f)
+		{
+			return res;
+		}
+
+		while(fgets(line, max_line_size, f))
+		{
+			res.emplace_back(std::string(line));
+		}
+
+		fclose(f);
+	}
+	else
+	{
+		::close(pipefds[0]);
+		dup2(pipefds[1], STDOUT_FILENO);
+		execvp("runsc", argv);
+		exit(1);
+	}
+
+	return res;
+}
+
+std::vector<std::string> engine::runsc_list()
+{
+	std::vector<std::string> sandboxes;
+
+	const char *argv[] = {
+		"runsc", 
+		"--root",
+		m_runsc_root_path.c_str(),
+		"list",
+		NULL
+	};
+
+	std::vector<std::string> output = runsc((char **)argv);
+
+	for(auto &line : output)
+	{
+		if(line.find("running") != std::string::npos)
+		{
+			std::string sandbox = line.substr(0, line.find_first_of(" ", 0));
+			sandboxes.emplace_back(sandbox);
+		}
+	}
+
+	return sandboxes;
+}
+
+void engine::runsc_trace_create(const std::string &sandbox_id, bool force)
+{
+	const char *argv[] = {
+		"runsc", 
+		"--root",
+		m_runsc_root_path.c_str(),
+		"trace",
+		"create",
+		force ? "--force" : "",
+		"--config", 
+		m_trace_session_config_path.c_str(),
+		sandbox_id.c_str(),
+		NULL
+	};
+
+	runsc((char **)argv);
```

Comment:
Just a stupid question, is there a way to check if the call fails? maybe a return value